src/features/manualDatasetCreation/state/manualDatasetCreationActions.ts

```diff
@@ -0,0 +1,72 @@
+import { ApiActionThunk, CALL_API } from "../../../store/api"
+import {
+  MANUAL_DATASET_CREATION_UPLOAD,
+  SET_MANUAL_DATASET_CREATION_META,
+  SET_MANUAL_DATASET_CREATION_README,
+  SET_MANUAL_DATASET_CREATION_COMMIT_TITLE,
+  SET_MANUAL_DATASET_CREATION_TITLE
+} from "./manualDatasetCreationState"
+import { Meta } from "../../../qri/dataset"
+
+export default function uploadFile () {
+  return ({
+    type: MANUAL_DATASET_CREATION_UPLOAD
+  })
+}
+
+export interface ManualDatasetCreationSetTextAction {
+  type: string
+  text: string
+}
+
+export interface ManualDatasetCreationSetMetaAction {
+  type: string
+  meta: Meta
+}
+
+export function setManualDatasetCreationCommitTitle (title: string): ManualDatasetCreationSetTextAction {
+  return {
+    type: SET_MANUAL_DATASET_CREATION_COMMIT_TITLE,
+    text: title
+  }
+}
+
+export function setManualDatasetCreationTitle (title: string): ManualDatasetCreationSetTextAction {
+  return {
+    type: SET_MANUAL_DATASET_CREATION_TITLE,
+    text: title
+  }
+}
+
+export function setManualDatasetCreationReadme (readme: string): ManualDatasetCreationSetTextAction {
+  return {
+    type: SET_MANUAL_DATASET_CREATION_README,
+    text: readme
+  }
+}
+
+export function setManualDatasetCreationMeta (meta: Meta): ManualDatasetCreationSetMetaAction {
+  return {
+    type: SET_MANUAL_DATASET_CREATION_META,
+    meta
+  }
+}
+
+export function saveManualDataset (bodyFile: File): ApiActionThunk {
+  return async (dispatch, getState) => {
+    return dispatch({
+      type: 'saveform',
```

Comment:
not necessary to change it, but just so you know if you want to match the backend, on the backend we now refer to this as 'saveupload', rather than 'saveform'